src/Datadog.Trace/Logging/LibLogScopeEventSubscriber.cs

```diff
@@ -32,6 +41,41 @@ internal class LibLogScopeEventSubscriber : IDisposable
 
         private bool _safeToAddToMdc = true;
 
+#if NETFRAMEWORK
+        static LibLogScopeEventSubscriber()
+        {
+            // Check if IIS automatic instrumentation has set the AppDomain property to indicate the PreStartInit state
+            // If the property is not set, we must rely on a different method of determining the state
+            object state = AppDomain.CurrentDomain.GetData(NamedSlotName);
+            if (state is bool boolState)
+            {
+                _performAppDomainFlagChecks = true;
+                _executingIISPreStartInit = boolState;
+            }
+            else
+            {
+                _performAppDomainFlagChecks = false;
```

Comment:
As this can have a performance impact, I believe we should add a warning log when `_performAppDomainFlagChecks` is false

Reply:
Sounds good, will add

Reply:
@kevingosse I made the change in c5c9c63da39cec5fd1863487a0d79c43e734d46b . Is that good for you?